src/atomate2/cp2k/sets/core.py

```diff
@@ -0,0 +1,309 @@
+
+import logging
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Any
+
+import numpy as np
+from pymatgen.core import Structure
+from pymatgen.core.periodic_table import Element
+from pymatgen.io.cp2k.inputs import Cp2kInput
+from pymatgen.io.cp2k.outputs import Cp2kOutput
+from pymatgen.io.cp2k.utils import get_truncated_coulomb_cutoff
+
+from atomate2.common.schemas.math import Vector3D
+from atomate2.cp2k.sets.base import Cp2kInputGenerator, multiple_input_updators
+
+logger = logging.getLogger(__name__)
+
+
+__all__ = [
+    "StaticSetGenerator",
+    "RelaxSetGenerator",
+    "CellOptSetGenerator",
+    "HybridStaticSetGenerator",
+    "HybridRelaxSetGenerator",
+    "HybridCellOptSetGenerator",
+    "NonSCFSetGenerator",
+]
+
+
+@dataclass
+class StaticSetGenerator(Cp2kInputGenerator):
+    """
+    Class to generate CP2K static input sets.
+
+    Parameters
+    ----------
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        Get updates to the input for a static CP2K job.
+
+        Parameters
+        ----------
+
+
+        Returns
+        -------
+        dict
+            A dictionary of updates to apply.
+        """
+        updates = {"run_type": "ENERGY_FORCE"}
+        return updates
+
+@dataclass
+class RelaxSetGenerator(Cp2kInputGenerator):
+    """
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        """
+        updates = {
+            "run_type": "GEO_OPT",
+            "activate_motion": {
+                'optimizer': "BFGS",
+                "trust_radius": 0.1
+            },
+        }
+        return updates
+
+@dataclass
+class CellOptSetGenerator(Cp2kInputGenerator):
+    """
+
+    """
+
+    def get_input_updates(self, *args, **kwargs) -> dict:
+        """
+        """
+        updates = {
+            "run_type": "CELL_OPT",
+            "activate_motion": {
+                'optimizer': "BFGS",
+                "trust_radius": 0.1
+            },
+        }
+
+        return updates
+
+
+@dataclass
+class HybridSetGenerator(Cp2kInputGenerator):
+
+    hybrid_functional: str = "PBE0"
+    screen_on_initial_p: bool = False
+    screen_p_forces: bool = False
+    eps_schwarz: float = 1e-7
+    eps_schwarz_forces: float = 1e-5
+
+    def get_input_updates(self, structure, *args, **kwargs) -> dict:
+        updates = {
+            "activate_hybrid": {
+                "hybrid_functional": self.hybrid_functional,
+                "screen_on_initial_p": self.screen_on_initial_p,
+                "screen_p_forces": self.screen_p_forces,
+                "eps_schwarz": self.eps_schwarz,
+                "eps_schwarz_forces": self.eps_schwarz_forces,
+            },
+        }
+        if hasattr(structure, "lattice"):
+            updates['activate_hybrid']['cutoff_radius'] = get_truncated_coulomb_cutoff(structure)
+
+        return updates
+
+@dataclass
+@multiple_input_updators()
```

Comment:
I see why you've done this and if you feel strongly about it then I'm happy to keep it. However, I think this will lead to confusion with users and it will be much clearer and not that much repetition to just write out the update values explicitly. You could define a single dictionary with the HSE parameters at the top of the file if want to avoid boilerplate (but personally I don't think even this is necessary).

For the VASP workflows, there has traditionally been a lot of confusion about why certain values are getting set. I worry that this implementation (which is technically very neat) would also cause confusion.

Reply:
While I like how clean this is (and spent a good few hours trying to figure out how to do it). I'll defer to readability instead of elegance.  

Will fix in next commit.